tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/UsernamePasswordIntegrationTests.cs

```diff
@@ -285,29 +291,41 @@ AuthenticationResult authResult
         }
 
-        private async Task<AuthenticationResult> GetAuthenticationResultWithAssertAsync(
-            LabResponse labResponse,
-            HttpSnifferClientFactory factory,
-            IPublicClientApplication msalPublicClient,
-            string federationMetadata,
-            Guid testCorrelationId)
-        {
-            AuthenticationResult authResult = await msalPublicClient
-                .AcquireTokenByUsernamePassword(s_scopes, labResponse.User.Upn, new NetworkCredential("", labResponse.User.GetOrFetchPassword()).SecurePassword)
-                .WithCorrelationId(testCorrelationId)
-                .WithFederationMetadata(federationMetadata)
-                .ExecuteAsync(CancellationToken.None)
-                .ConfigureAwait(false);
-
-            Assert.IsNotNull(authResult);
-            Assert.AreEqual(TokenSource.IdentityProvider, authResult.AuthenticationResultMetadata.TokenSource);
-            Assert.IsNotNull(authResult.AccessToken);
-            Assert.IsNotNull(authResult.IdToken);
-            Assert.IsTrue(string.Equals(labResponse.User.Upn, authResult.Account.Username, StringComparison.InvariantCultureIgnoreCase));
-            AssertTelemetryHeaders(factory, false, labResponse);
-            AssertCCSRoutingInformationIsSent(factory, labResponse);
-
-            return authResult;
+        private async Task<AuthenticationResult> GetAuthenticationResultWithAssertAsync(
+            LabResponse labResponse,
+            HttpSnifferClientFactory factory,
+            IPublicClientApplication msalPublicClient,
+            string federationMetadata,
+            Guid testCorrelationId)
+        {
+            AuthenticationResult authResult = await msalPublicClient
+                .AcquireTokenByUsernamePassword(s_scopes, labResponse.User.Upn, new NetworkCredential("", labResponse.User.GetOrFetchPassword()).SecurePassword)
+                .WithCorrelationId(testCorrelationId)
+                .WithFederationMetadata(federationMetadata)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            Assert.IsNotNull(authResult);
+            Assert.AreEqual(TokenSource.IdentityProvider, authResult.AuthenticationResultMetadata.TokenSource);
+            Assert.IsNotNull(authResult.AccessToken);
+            Assert.IsNotNull(authResult.IdToken);
+            Assert.IsTrue(string.Equals(labResponse.User.Upn, authResult.Account.Username, StringComparison.InvariantCultureIgnoreCase));
+            AssertTelemetryHeaders(factory, false, labResponse);
+            AssertCCSRoutingInformationIsSent(factory, labResponse);
+
+            return authResult;
+        }
+
+        
+        private void AssertTenantProfiles(IDictionary<string, ITenantProfile> tenantProfiles, string tenantId)
+        {
+            Assert.IsNotNull(tenantProfiles);
```

Comment:
i would separate the idtoken claims, as a separate API. it's not easily found, and the reason for having them is separate from having the tenant profiles.

Reply:
@jennyf19 what do you mean by having a separate API for token claims? Not have them in the tenant profiles? 

I think having them in the tenant profiles as a dictionary is fine since the claims are for an id token. Also, point of having claims as a dictionary here is to get all the claims that exists in an id token and those can vary.

Aside, we already have [IdToken.cs](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/698f0fd1b0daba783c9028e099595f10a0d52da2/src/client/Microsoft.Identity.Client/Internal/IdToken.cs#L9) which we use as an internal class for IdTokenCacheItems.

cc: @jmprieur

Reply:
see earlier conversation where @jmprieur provided code to get the IdTokenClaims separately.

Reply:
Have changed the API, it now looks like this: 

```csharp
var authResult = await interactiveBuilder.ExecuteAsync().ConfigureAwait(false);
ClaimsPrincipal idTokenClaims = authResult.ClaimsPrincipal; 

var accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
foreach (var acc in accounts)
{
    Console.WriteLine($"Account for {acc.Username}");
    foreach (var tp in acc.GetTenantProfiles())
    {
        Console.WriteLine($"Tenant Profile in tenant {tp.TenantId} - claims {tp.ClaimsPrincipal.Claims.Count()} - is home tenant? {tp.IsHomeTenant}")
    }
}
```